sdk/ios/Configsum/Configsum/Configsum/Networking/ResultHandler.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  ResultHandler.swift
+//  Configsum
+//
+//  Created by Alexandru Savu on 2017-10-07.
+
+import Foundation
+
+enum ParseError<T, U>: Error {
+    case InvalidType(have: T, want: U)
+    case InvalidKey(T)
+}
+
+internal struct ResultHandler<T> {
+    internal let key: String
+    internal let persistence: Persistence
+    
+    internal init(persistence: Persistence,
+                  key: String = "") {
+        self.persistence = persistence
```

Comment:
any particular reason for add an empty default value for `key`, you are using the key in this class so I assume it cannot be nil or empty. If that is the case can you remove the default value & force the user to specify a value for `key` in the constructor

Reply:
it's because I'm using this structure both with and without key. once for fetching a config (no need for key) and once for retrieving a value ( key is needed )

Reply:
you could have used two constructors (with & without key) because there won't be autocompletion for params with a default value, it might cause confusion.

this is an internal class so I guess it is not a pressing issue :)

Reply:
good point